### PR TITLE
davix: fix on darwin

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/davix/default.nix [hunk 1]
```diff
@@ -36,8 +33,10 @@ stdenv.mkDerivation rec {
     openssl
     libxml2
     boost
-    libuuid
-  ] ++ lib.optional (defaultToLibcurl && !useEmbeddedLibcurl) curl
+    curl
+    Security
+  ]
```
Comment (ShamrockLee) [inline pkgs/tools/networking/davix/default.nix:38]:
```suggestion
  ]
  ++ lib.optional stdenv.isDarwin Security
```

**Review by ShamrockLee — CHANGES_REQUESTED** [review]
Thanks a lot for this speedy fix!

The Darwin-specific things need to be added conditional-wise, though.

**Comment by ShamrockLee** [comment]
Do we need to add a line in the release note about the input argument removal, or the input arguments of packages are considered to be packaging details and subject to changes?

**Comment by veprbl** [comment]
The option was useless in the first place, so I'd assume no one will be affected. The changelog is targeting nixos mainly.

**Review by ShamrockLee — APPROVED** [review]

**Pull request merged**